edb/edgeql/tracer.py

```diff
@@ -368,28 +416,31 @@ def trace_Path(node: qlast.Path, *,
                     tip = ctx.objects[ctx.path_prefix]
                 else:
                     # We can't reason about this path.
-                    return
+                    return None
 
             if step.type == 'property':
                 if ptr is None:
                     # This is either a computable def, or
                     # unknown link, bail.
-                    return
-
-                lprop = ptr.getptr(ctx.schema, step.ptr.name)
-                if lprop is None:
-                    # Invalid link property reference, bail.
-                    return
-
-                if isinstance(lprop, Pointer):
-                    src = lprop.source
-                    src_name = src.get_name(ctx.schema)
-                    if isinstance(src, Pointer) and src.source is not None:
-                        src_src_name = src.source.get_name(ctx.schema)
-                        source_name = f'{src_src_name}@{src_name}'
-                    else:
-                        source_name = src_name
-                    ctx.refs.add(f'{source_name}@{step.ptr.name}')
+                    return None
+
+                elif isinstance(ptr, (s_links.Link, Pointer)):
```

Comment:
@elprans do we have this weird union type here?

Reply:
Yes.  When we trace SDL to extract the dependency graph we use makeshift standins for schema objects that are found in the SDL file being analyzed.  Existing schema objects would be regular schema objects, hence this union.

Reply:
Alright. Is this PR ready?

Reply:
I've added some cosmetic changes to imports. I'm trying to avoid name clashes between `typing` and the local classes, so I import specific items from `typing` rather than `from typing import *` or `import typing`. Selectively importing items from typing is more consistent with `from typing import *` used everywhere else in the project.